user_interface.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <string>
+#include "user_interface.h"
+#include <cmath>
+#include <iomanip> //std::setprecision
+
+using namespace std;
+//const ostream &standard = std::cout;
+const string white_line = "\011\012\013\014\015\040";
+
+void print_greeting(ostream &out)
+{
+    out << "Welcome to the random number generator!\n"
+        << "It was done as a laboratory programming job.\n";
+}
+
+string clear_whites(const string &s){
+    size_t first = s.find_first_not_of(white_line);
+
+    if(first == string::npos) return "";
+    size_t last = s.find_last_not_of(white_line, s.size() );
+    return s.substr(first, last - first + 1);
+}
+
+int choose_gen(ostream &out)
+{
+    bool correct_ans = false;
```

Comment:
There is no need to use this bool flag if you do an infinite loop and just return the  result if the input is valid.
```suggestion
```